packages/react-storage/src/components/StorageBrowser/views/LocationActionView/DestinationPicker.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { PaginateControl } from '../../views/Controls/Paginate';
+import {
+  EmptyMessageControl,
+  LoadingControl,
+  MessageControl,
+} from '../Controls';
+import { ViewElement } from '../../context/elements';
+import { displayText } from '../../displayText/en';
+import { useDestinationPicker } from './CopyView/useDestinationPicker';
+import { CLASS_BASE } from '../constants';
+import { DataTableControl } from '../../controls/DataTableControl';
+import { ControlsContextProvider } from '../../controls/context';
+import {
+  getDestinationListFullPrefix,
+  getDestinationPickerTableData,
+} from './utils/getDestinationPickerDataTable';
+import { ControlsContext } from '../../controls/types';
+import { Breadcrumb } from '../../components/BreadcrumbNavigation';
+import { DescriptionList } from '../../components/DescriptionList';
+const {
+  actionSetDestination,
+  actionDestinationPickerCurrentFolderSelected,
+  actionDestinationPickerNoMoreFolders,
+  actionDestinationPickerDefaultError,
+} = displayText;
+
+const DEFAULT_PAGE_SIZE = 10;
+export const DEFAULT_LIST_OPTIONS = {
+  pageSize: DEFAULT_PAGE_SIZE,
+  delimiter: '/',
+};
+
+export const DestinationPicker = ({
+  destinationList,
+  onSetDestinationList,
+}: {
+  destinationList: string[];
+  onSetDestinationList: (destination: string[]) => void;
+}): React.JSX.Element => {
+  const {
+    bucket,
+    items,
+    hasNextToken,
+    currentPage,
+    isLoading,
+    hasError,
+    handleNext,
+    handlePrevious,
+    range,
+  } = useDestinationPicker({ destinationList });
+
+  const handleNavigateFolder = (key: string) => {
+    const newPath = [...destinationList, key.replace('/', '')];
+    onSetDestinationList(newPath);
+  };
+
+  const handleNavigatePath = (index: number) => {
+    const newPath = destinationList.slice(0, index + 1);
+    onSetDestinationList(newPath);
+  };
+
+  const pageItems = React.useMemo(() => {
+    const [start, end] = range;
+    return items.slice(start, end);
+  }, [range, items]);
+
+  const disableNext =
+    !hasNextToken && currentPage * DEFAULT_PAGE_SIZE > items.length;
+  const disablePrevious = currentPage === 1;
+
+  const tableData = getDestinationPickerTableData({
+    items: pageItems,
+    handleNavigateFolder,
+  });
+
+  const contextValue: ControlsContext = {
+    data: {
+      tableData,
+    },
+  };
+
+  const noSubfolders = !items.length;
+  const messageVariant = noSubfolders ? 'info' : 'error';
+  const message = noSubfolders
+    ? `${actionDestinationPickerCurrentFolderSelected} ${getDestinationListFullPrefix(
+        destinationList
+      )}. ${actionDestinationPickerNoMoreFolders}`
+    : actionDestinationPickerDefaultError;
```

Comment:
Need to rethink the approach here, this is too limited and complicated for granular overrides

Reply:
🍕

Reply:
⭐ suck

Reply:
@reesscot was busy feeding 🍕